packages/testing/src/extension.ts

```diff
@@ -146,22 +150,30 @@ export const it = vitestIt.extend<{
       await amqpChannel.assertQueue(queue);
       await amqpChannel.bindQueue(queue, exchange, routingKey);
 
-      return (nbEvents = 1) =>
-        new Promise((resolve) => {
-          const messages: amqpLib.ConsumeMessage[] = [];
-          amqpChannel.consume(
-            queue,
-            (msg) => {
-              if (msg) {
-                messages.push(msg);
-                if (messages.length >= nbEvents) {
-                  resolve(messages);
-                }
-              }
-            },
-            { noAck: true },
-          );
-        });
+      const messages: amqpLib.ConsumeMessage[] = [];
+      await amqpChannel.consume(
+        queue,
+        (msg) => {
+          if (msg) {
+            messages.push(msg);
+          }
+        },
+        { noAck: true },
+      );
+
+      return async (nbEvents = 1) => {
```

Comment:
transform it into an options object with neEvents and timeout. options will be optional so nbEvents and timeout will have default values